src/os/windows.rs

```diff
@@ -0,0 +1,48 @@
+//! Windows platform support.
+
+use std::fs::File;
+use std::io::Result;
+use std::os::windows::io::AsRawHandle;
+
+use windows::Win32::Foundation::*;
+use windows::Win32::Storage::FileSystem::*;
+use windows::Win32::System::IO::*;
+
+/// Fills a buffer from the given file at the given offset.
+///
+/// This command fills the entire buffer unless at EOF.
+pub fn fill_buffer(file: &File, buffer: &mut [u8], offset: u64) -> Result<usize> {
+    let mut read = 0;
+    while read < buffer.len() {
+        let offset = offset + u64::try_from(read).unwrap();
+        let mut overlapped = OVERLAPPED {
+            Anonymous: OVERLAPPED_0 {
+                Anonymous: OVERLAPPED_0_0 {
+                    Offset: (offset & 0xFFFFFFFF) as u32,
+                    OffsetHigh: (offset >> 32) as u32,
+                },
+            },
```

Comment:
Oh come on, Windows. Might be worthwhile to add a pointer to a doc like [this one](https://learn.microsoft.com/en-us/windows/win32/api/minwinbase/ns-minwinbase-overlapped) or just a prose description that we have to split up the 64-bit offset like this.